etc/afpd/file.c

```diff
@@ -2243,7 +2243,7 @@ int afp_exchangefiles(AFPObj *obj, char *ibuf, size_t ibuflen _U_, char *rbuf _U
      * NOTE: the temp file will be in the dest file's directory. it
      * will also be inaccessible from AFP. */
     memcpy(temp, APPLETEMP, sizeof(APPLETEMP));
-    if (!mktemp(temp)) {
+    if ( (mkstemp(temp)) == -1) {
         err = AFPERR_MISC;
```

Comment:
This leaks a file descriptor.

Reply:
Good catch! See if this change resolves the issue properly (I basically looked at the 3.1 branch and copied what was done there.)